### ISSUE TITLE
Add validation to GUI inputs

### ISSUE THREAD

**Issue by netniV** [Cacti/cacti#1442]
Since we have a lot of validation applied to the URI's to ensure that they have the correct values, this should also be taken a stage further and be applied to the input fields.  If an input isn't correct, the input should show red around it.  

At the moment, for example, it's possible to save an alpha in a numeric field (according to the GUI) but I'm pretty sure if the backend field is not an alpha it will fail silently (using the bug #1412 _Save is reported as successful even though SQL statement fails_ ).

However, on screen, the GUI shows the field as edited with no clue that it has a problem.

I think the simple way to fix this is to update the field definitions to include a 'validation' entry that declares the type, then have a jQuery leave even that performs that validation.  This is obviously only needed for textbox entries as textarea's shoudl be alphanumeric, and combobox's are fixed.

**Comment by cigamit** [comment]
Well, that is what form_input_validate() and is_error() are for.  If there is an error in the initial validation, then the raise_message() is supposed to set an error and then highlight the form element that is incorrect.  So, in this case, the form_input_validate() must not be catching there error.  It's not pretty, but it does the job.

**Comment by netniV** [comment]
Yeah, I just think if we are moving the GUI forward, it really should try and catch some of these things before the user hits save to prevent potential errors.

I was also thinking that this is probably a longer term enhancement, but the advantage being that if it exists within the field definition, we can also then simplify things for URI parameters by simply saying that parameter is that field, thus gets the exact same validation without relying on needing to remember.

**Comment by cigamit** [comment]
Yes, longer term.

**Comment by cigamit** [comment]
Let's find a form validation library, and then add a type to the form arrays, like 'phone_number', 'email', 'integer', etc.  So long as the validation api takes a proprietary tag in the &lt;input&gt;, then we should be all set.  I'll take a quick look.

**Comment by cigamit** [comment]
http://jqueryvalidation.org is one simple MIT licensed example.
